applications/MeshingApplication/custom_utilities/uniform_refine_utility.cpp

```diff
@@ -0,0 +1,680 @@
+//    |  /           |
+//    ' /   __| _` | __|  _ \   __|
+//    . \  |   (   | |   (   |\__ `
+//   _|\_\_|  \__,_|\__|\___/ ____/
+//                   Multi-Physics
+//
+//  License:		 BSD License
+//					 Kratos default license: kratos/license.txt
+//
+//  Main authors:    Miguel Maso Sotomayor
+//
+
+
+// System includes
+
+
+// External includes
+
+
+// Project includes
+#include "includes/define.h"
+#include "includes/variables.h"
+#include "uniform_refine_utility.h"
+
+
+namespace Kratos
+{
+/// Default constructor
+template< unsigned int TDim>
+UniformRefineUtility<TDim>::UniformRefineUtility(ModelPart& rModelPart, int RefinementLevel) :
+    mrModelPart(rModelPart),
+    mFinalRefinementLevel(RefinementLevel),
+    mSubModelPartsColors(mrModelPart)
+{
+    // Initialize the member variables storing the Id
+    mLastNodeId = 0;
+    mLastElemId = 0;
+    mLastCondId = 0;
+
+    // Get the last node id
+    const int nnodes = mrModelPart.Nodes().size();
+    for (int i = 0; i < nnodes; i++)
+    {
+        ModelPart::NodesContainerType::iterator inode = mrModelPart.NodesBegin() + i;
+        if (inode->Id() > mLastNodeId)
+            mLastNodeId = inode->Id();
+    }
+
+    // Get the elements id
+    const int n_elements = mrModelPart.Elements().size();
```

Comment:
This is SizeType->std:size_t

Reply:
Idem for all the cases, use int for the omp loops, but remember to use static_cast in that case

Reply:
Thank you, I am going to apply those changes